src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs

```diff
@@ -575,5 +582,22 @@ private void PopulateProjectAndPackageReferences(PackageBuilder packageBuilder,
                 }
             }
         }
+
+        private static IDictionary<string, string> ParsePropertiesAsDictionary(string[] properties)
+        {
+            var dictionary = new Dictionary<string, string>();
+            foreach (var item in properties)
+            {
+                string[] pair = item.Split('=');
+                if (pair.Length != 2)
```

Comment:
@alpaix @emgarten wrote the tests and made changes as per feedback. all edge cases should be handled now. One thing to note is that msbuild automatically strips out the double quotes so we don't need to do that ourselves